### ISSUE TITLE
RocksDB public get and puts only accepts complete byte arrays

### ISSUE THREAD

**Issue by Zelldon** [zeebe-io/zeebe#1205]
RocksDB public put and get methods accept only byte arrays.
This means you can't use ExpandableBuffer for the key or values, which means
you have to allocate byte arrays for the values and keys, if they can have arbitrary length.
For sure if the length is always the same, then there is no problem in pre-allocating and reusing the byte arrays.

Currently for deployments we allocate every time an new byte array to put or get something from RocksDB, since it is not in the hot path. If we want to change the state management in the workflow processing, then we need a solution which is more or less garbage free.

There exist an open PR https://github.com/facebook/rocksdb/pull/2283 to support ByteBuffers, but it is already open since an year.

Alternative way would be to use reflection, to call the internal `protected native void put(long handle, byte[] key, int keyOffset,
      int keyLength, byte[] value, int valueOffset, int valueLength)` [see here](https://github.com/facebook/rocksdb/blob/master/java/src/main/java/org/rocksdb/RocksDB.java#L2276)

Could look like this
```java

  private long nativeHandle_;
  private Method put;
  [...]
        final Field declaredField = RocksDB.class.getSuperclass().getDeclaredField("nativeHandle_");
        declaredField.setAccessible(true);
        this.nativeHandle_ = (long) declaredField.get(db);
        put =
            RocksDB.class.getDeclaredMethod(
                "put",
                Long.TYPE,
                byte[].class,
                Integer.TYPE,
                Integer.TYPE,
                byte[].class,
                Integer.TYPE,
                Integer.TYPE);
        put.setAccessible(true);

  [...]

  private void put(
      byte[] key, int keyOffset, int keyLength, byte[] value, int valueOffset, int valueLength)
      throws Exception {
    put.invoke(db, nativeHandle_, key, keyOffset, keyLength, value, valueOffset, valueLength);
  }
```

blocks #1178 
blocks #1188

**Comment by Zelldon** [comment]
blocks #1178 and blocks #1188

**Comment by Zelldon** [comment]
created an pull request for this https://github.com/facebook/rocksdb/pull/4302 there is also another pull request already open https://github.com/facebook/rocksdb/pull/2303

**Comment by npepinpe** [comment]
I think this will be addressed in #1195, so I would close this unless anyone objects.
